File: platopy/nlgen/rule_set.py
rule_set = {
    'none': 'rules.one',
    'all': 'rules.all',
    'pronouns': 'rules.pro',
    'ellipses': 'rules.ellip',
    'aggregation': 'rules.agg',
    'modification': 'rules.mod',
    'partitives': 'rules.part',
    'connectives': 'rules.conn'
}

class RuleSet:
    def __init__(self, display_name, filename):
        self.display_name = display_name
        self.filename = filename

    def set_filename(self, filename):
        self.filename = filename

    def get_filename(self):
        return self.filename

    def __str__(self):
        return self.display_name

File: platopy/nlgen/RuleSet.java
package openproof.plato.nlgen;

public enum RuleSet {
	
	// Enum for ruleset types, each recording its corresponding Lingo filename argument.
	
	NONE("None", "rules.one"),
	ALL("All", "rules.all"),
	PRONOUNS("Pronouns", "rules.pro"),
	ELLIPSES("Ellipses", "rules.ellip"),
	AGGREGATION("Aggregation", "rules.agg"),
	MODIFICATION("Modification", "rules.mod"),
	PARTITIVES("Partitives", "rules.part"),
	CONNECTIVES("Connectives", "rules.conn");
	
	private String displayName;
	private String filename;

	RuleSet(String s, String filename) {
		this.displayName = s;
		this.setFilename(filename);
	}
	
	public String toString() { return this.displayName; }

	public String getFilename() {
		return filename;
	}

	public void setFilename(String filename) {
		this.filename = filename;
	}

}
